pkg/app/app.go

```diff
@@ -282,8 +282,11 @@ func CreateApp(args Args, regs []registries.Registry) App {
 		KeepNamespaceOnError: app.config.GetBool("openshift.keep_namespace_on_error"),
 	}
 	apb.InitializeClusterConfig(clusterConfig)
+	brokerNS := app.config.GetString("openshift.namespace")
+	// initialize the work factory
+	workFactory := broker.NewWorkFactory()
```

Comment:
I'm not sure of the need for the `WorkFactory`.

Reply:
The intention here was creating an interface that would encapsulate the creation of work and allow a mock to be used later down the line to do tests against a running broker without it needing to actually start pods etc. Initializing the stuct directly in the code makes it difficult to control the behavior. Really just saw it as a step in the direction of a more unit testable broker.